### DISCUSSION TITLE
starting view with HTML comment breaks component

### DISCUSSION THREAD

**Discussion by felixsc** [livewire/livewire#4152] [7. Bugs (archived)] [answered]
Hello there!

**Issue:**
Component breaks when starting a livewire view with an HTML comment. 

**Description:**
If the `livewire-component.blade.php` view starts with a simple HTML comment, it breaks the component (see screenshot).

Now, I know that views should only have one child element, but I'm not sure if comments should count towards that nor not. Or if this is even the issue in this case.

I've come across this using TailwindUI, which often adds a comment at the top of a component. It's not a big deal to remove the comment, but I wasn't sure if this counts as a bug or not.

Either way, starting the file with an HTML comment, seems to break the component.

**Screenshot:**
This is the result
![3-11-2021_Za40SzLFKA](https://user-images.githubusercontent.com/2606794/140086709-586d31c6-be83-45f4-9b28-1760501f81e9.jpeg)

**Example:**
Here is an example. The only thing I've changed is the `livewire-component.blade.php` the rest is just livewire boilerplate.
It includes an empty div and a comment.

https://laravelplayground.com/#/snippets/a21eba6a-496d-4e47-98b7-e6dbe08d3331

The comment breaking it:
`<!-- This is just a comment -->`

Same empty component without comment:
https://laravelplayground.com/#/snippets/6bc65f0f-d0c2-4732-80a3-502fae670742

**Comment by joshhanley** [comment, marked as answer]
@felixsc html comments are actually valid elements, so you should consider them as part of single root element in a view thing.

But we did actually add some checks in for comments in v2.6, so comments should be skipped now, see this PR #3289

Can you confirm what version of Livewire you are using?

Also I don't know what version of Livewire that the Laravel Playground is using, so I'd assume it is not up to date.

If you could test it locally that would be great! 🙂

Hope this helps!

**Reply by felixsc** [reply]
Thanks, Josh!

Yes, I was on 2.5.5 and updating to the latest version fixed it 👍

Laravel Playground seems to use an older version as well.

Didn't know that comments are treated as regular DOM elements, so TIL 😁 

Thanks for taking the time. Have a great day!

**Reply by joshhanley** [reply]
No worries! 🙂
